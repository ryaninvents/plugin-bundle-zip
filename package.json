{
  "name": "@ryaninvents/plugin-bundle-zip-node",
  "version": "0.0.0-semantically-released",
  "description": "Plugin for @pika/pack to zip built files for Node.js",
  "author": "Ryan Kennedy <ryaninvents@users.noreply.github.com>",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "gulp": "^4.0.0",
    "gulp-zip": "^4.2.0"
  },
  "@pika/pack": {
    "pipeline": [
      [
        "@pika/plugin-standard-pkg"
      ],
      [
        "@pika/plugin-build-node",
        {
          "minNodeVersion": "8"
        }
      ]
    ]
  },
  "devDependencies": {
    "@pika/pack": "^0.3.1",
    "@pika/plugin-build-node": "^0.3.10",
    "@pika/plugin-standard-pkg": "^0.3.10",
    "@pika/types": "^0.3.1",
    "@semantic-release/changelog": "^3.0.2",
    "@semantic-release/commit-analyzer": "^6.1.0",
    "@semantic-release/git": "^7.0.8",
    "@semantic-release/github": "^5.2.10",
    "@semantic-release/npm": "^5.1.4",
    "@semantic-release/release-notes-generator": "^7.1.4",
    "@types/gulp": "^4.0.5",
    "@types/gulp-zip": "^4.0.0",
    "@types/node": "^10.12.24",
    "eslint": "^5.13.0",
    "eslint-config-semistandard": "^13.0.0",
    "eslint-config-standard": "^12.0.0",
    "eslint-plugin-import": "^2.16.0",
    "eslint-plugin-node": "^8.0.1",
    "eslint-plugin-promise": "^4.0.1",
    "eslint-plugin-standard": "^4.0.0",
    "semantic-release": "^15.13.3",
    "typescript": "^3.3.3"
  },
  "eslintConfig": {
    "extends": "semistandard"
  },
  "release": {
    "branch": "develop",
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/changelog",
      [
        "@semantic-release/npm",
        {"pkgRoot": "./pkg"}
      ],
      "@semantic-release/github",
      [
        "@semantic-release/git",
        {
          "assets": [
            "CHANGELOG.md"
          ],
          "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
        }
      ]
    ]
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "pack build",
    "semantic-release": "semantic-release"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/ryaninvents/plugin-bundle-zip.git"
  }
}
