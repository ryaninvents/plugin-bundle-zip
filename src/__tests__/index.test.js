import fs from 'fs-extra';
import { join, resolve } from 'path';
import { dir as createTempDir, file as createTempFile } from 'temporarily';
import del from 'del';
import execa from 'execa';
import archive from 'ls-archive';
import { name as thisPackageName } from '../../package.json';

const DEFAULT_PKG_JSON = {
  name: 'test-pkg',
  private: true,
  version: '0.0.0-develop',
  scripts: {
    build: 'pack build'
  },
  '@pika/pack': {
    'pipeline': [
      [
        '@pika/plugin-standard-pkg'
      ],
      [
        '@pika/plugin-build-node',
        {
          'minNodeVersion': '8'
        }
      ],
      [
        thisPackageName
      ]
    ]
  },
  'devDependencies': {
    '@pika/pack': '^0.3.1',
    '@pika/plugin-build-node': '^0.3.10',
    '@pika/plugin-standard-pkg': '^0.3.10',
    [thisPackageName]: resolve(__dirname, '../../pkg')
  }
};

describe('@ryaninvents/plugin-bundle-dependencies', () => {
  let createdDirs = [];

  async function initRepo ({ packageJson: pkg = DEFAULT_PKG_JSON, packageManager = 'npm', stdio } = {}) {
    const { filepath: workingDir } = createTempDir(
      { name: 'test-repo-{wwwwdddd}' },
      [createTempDir({ name: 'src' }, [
        createTempFile({ name: 'index.js', data: 'console.log("Hello world");' })
      ])]
    );
    let packageJson = pkg;
    if (typeof packageJson === 'function') {
      packageJson = packageJson(workingDir);
    }
    await fs.writeFile(join(workingDir, 'package.json'), JSON.stringify(packageJson, null, 2));
    await execa('npm', ['install'], {
      cwd: workingDir,
      env: { NODE_ENV: 'development' }
    });
    createdDirs.push(workingDir);
    return { workingDir };
  }

  afterEach(async () => {
    await del(createdDirs, { force: true });
    createdDirs = [];
  });

  it('should create a zip file with contents of dist-node', async () => {
    const { workingDir } = await initRepo({
      packageJson: {
        ...DEFAULT_PKG_JSON,
        dependencies: {
          // Use a couple of packages known to basically be one-liners
          // in order to test quickly
          'is-sorted': 'latest',
          'map-obj': 'latest'
        }
      },
      stdio: 'inherit'
    });
    await execa('npm', ['run', 'build'], {
      env: { NODE_ENV: 'production' },
      cwd: workingDir
    });
    expect(async () => fs.access(`${workingDir}/pkg/package.json`, fs.constants.F_OK))
      .not.toThrow();
    expect(async () => fs.access(`${workingDir}/pkg/dist-node.zip`, fs.constants.F_OK))
      .not.toThrow();
    const archiveEntries = await new Promise((resolve, reject) => {
      archive.list(`${workingDir}/pkg/dist-node.zip`, (err, results) => {
        if (err) return reject(err);
        return resolve(results.map(file => file.getPath()).sort());
      });
    });
    expect(archiveEntries).toMatchSnapshot();
  }, 60e3);

  it('should honor "preserve" option', async () => {
    const { workingDir } = await initRepo({
      packageJson: {
        ...DEFAULT_PKG_JSON,
        '@pika/pack': {
          'pipeline': [
            [
              '@pika/plugin-standard-pkg'
            ],
            [
              '@pika/plugin-build-node',
              {
                'minNodeVersion': '8'
              }
            ],
            [
              thisPackageName,
              { preserve: true }
            ]
          ]
        },
        dependencies: {
          // Use a couple of packages known to basically be one-liners
          // in order to test quickly
          'is-sorted': 'latest',
          'map-obj': 'latest'
        }
      },
      stdio: 'inherit'
    });
    await execa('npm', ['run', 'build'], {
      env: { NODE_ENV: 'production' },
      cwd: workingDir
    });
    expect(async () => fs.access(`${workingDir}/pkg/package.json`, fs.constants.F_OK))
      .not.toThrow();
    expect(async () => fs.access(`${workingDir}/pkg/dist-node.zip`, fs.constants.F_OK))
      .not.toThrow();
    const archiveEntries = await new Promise((resolve, reject) => {
      archive.list(`${workingDir}/pkg/dist-node.zip`, (err, results) => {
        if (err) return reject(err);
        return resolve(results.map(file => file.getPath()).sort());
      });
    });
    expect(archiveEntries).toMatchSnapshot();
  }, 60e3);

  it('should honor "directory" and "bundleName" options', async () => {
    const { workingDir } = await initRepo({
      packageJson: {
        ...DEFAULT_PKG_JSON,
        '@pika/pack': {
          'pipeline': [
            [
              '@pika/plugin-standard-pkg'
            ],
            [
              '@pika/plugin-build-node',
              {
                'minNodeVersion': '8'
              }
            ],
            [
              thisPackageName,
              { directory: 'dist-src', bundleName: 'src' }
            ]
          ]
        },
        dependencies: {
          // Use a couple of packages known to basically be one-liners
          // in order to test quickly
          'is-sorted': 'latest',
          'map-obj': 'latest'
        }
      },
      stdio: 'inherit'
    });
    await execa('npm', ['run', 'build'], {
      env: { NODE_ENV: 'production' },
      cwd: workingDir
    });
    expect(async () => fs.access(`${workingDir}/pkg/package.json`, fs.constants.F_OK))
      .not.toThrow();
    expect(async () => fs.access(`${workingDir}/pkg/dist-src.zip`, fs.constants.F_OK))
      .not.toThrow();
    const archiveEntries = await new Promise((resolve, reject) => {
      archive.list(`${workingDir}/pkg/dist-src.zip`, (err, results) => {
        if (err) return reject(err);
        return resolve(results.map(file => file.getPath()).sort());
      });
    });
    expect(archiveEntries).toMatchSnapshot();
  }, 60e3);
});
