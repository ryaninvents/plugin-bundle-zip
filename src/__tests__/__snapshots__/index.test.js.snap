// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@ryaninvents/plugin-bundle-dependencies should create a zip file with contents of dist-node 1`] = `
Array [
  "index.js",
]
`;

exports[`@ryaninvents/plugin-bundle-dependencies should honor "directory" and "bundleName" options 1`] = `
Array [
  "index.js",
]
`;

exports[`@ryaninvents/plugin-bundle-dependencies should honor "preserve" option 1`] = `
Array [
  "dist-node",
  "dist-node/index.js",
  "dist-src",
  "dist-src/index.js",
  "package.json",
]
`;
